### PR TITLE
Autodoc generation for cobra commands

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
# TL;DR

- **Get execution**

Screen Shot 2021-02-11 at 1 24 57 PM

- **Get project**
Screen Shot 2021-02-11 at 1 29 26 PM

- **Get workflow**
Screen Shot 2021-02-11 at 1 29 43 PM

- Update project
Screen Shot 2021-02-11 at 1 29 56 PM

- Register files
Screen Shot 2021-02-11 at 1 30 07 PM

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?
Pending items
Windows support that requires changes in make.bat file.

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: cmd/get/get.go [hunk 1]
```diff
@@ -14,11 +14,17 @@ func CreateGetCommand() *cobra.Command {
...
-		"workflow":   {CmdFunc: getWorkflowFunc, Aliases: []string{"workflows"}},
-		"launchplan": {CmdFunc: getLaunchPlanFunc, Aliases: []string{"launchplans"}},
-		"execution":  {CmdFunc: getExecutionFunc, Aliases: []string{"executions"}},
+		"project": {CmdFunc: getProjectsFunc, Aliases: []string{"projects"}, ProjectDomainNotRequired: true,
+			Short: "Gets project resources",
+			Long:  "Retrieves all the projects"},
+		"task":       {CmdFunc: getTaskFunc, Aliases: []string{"tasks"},Short: "Gets task resources",
+			Long:  "Retrieves all the tasks"},
+		"workflow":   {CmdFunc: getWorkflowFunc, Aliases: []string{"workflows"}, Short: "Gets workflow resources",
+			Long:  "Retrieves all the workflows"},
```
Comment (kumare3) [inline cmd/get/get.go:23]: you did not move the Long to a variable in every file that containes for example the `getLaunchPlanFunc` that would set a very nice precedent?

**Comment by kumare3** [comment]
you have not checked in all the `gen` files. I think we will have to unless you `make html -> depend on -> make gen-docs` can we do that?

File: cmd/get/get.go (hunk 1 above)
Comment (pmahindrakar-oss) [reply cmd/get/get.go]: Done. Moved these

Comment (pmahindrakar-oss) [reply cmd/get/get.go]: Added a few examples.

**Comment by pmahindrakar-oss** [comment]
> you have not checked in all the `gen` files. I think we will have to unless you `make html -> depend on -> make gen-docs` can we do that? But, wait we cannot do that, because golang wont be available in readthedocs

I agree we cannot add that dependency . I have added the gen files in the PR.

File: cmd/get/get.go [hunk 2]
```diff
@@ -6,19 +6,197 @@ import (
...
+// Long descriptions are whitespace sensitive when generating docs using sphinx.
+const (
+	getCmdShort = `Used for fetching various flyte resources including tasks/workflows/launchplans/executions/project.`
+	getCmdLong  = `
+Example get projects.
+::
+
+ bin/flytectl get project
+`
+	projectShort = "Gets project resources"
```
Comment (kumare3) [inline cmd/get/get.go]: I was thinking of adding this to the project file

**Comment by kumare3** [comment]
Awesome!

**Review by kumare3 — APPROVED** [review]
